### PR TITLE
Include python 3.8

### PR DESCRIPTION
**Pull request by 10ourto**
Nice library. I need to use it with python 3.8, unfortunately can't bump python version for a while. I've included python 3.8 to library dependencies seems there is no any blocker for that.

### CODE REVIEW

**Comment by 10ourto** [comment]
@arthurio same motivation as in https://github.com/arthurio/fastapi-filter/pull/114 but for python 3.8

File: pyproject.toml [hunk 1]
```diff
@@ -38,16 +38,17 @@ classifiers = [
...
     "Programming Language :: Python :: 3.11",
     "Programming Language :: Python :: 3 :: Only",
 ]
 
 [tool.poetry.dependencies]
-python = ">=3.9,<3.12"
+python = ">=3.8,<3.12"
 fastapi = ">=0.78,<0.86"
-pydantic = ">=1.10.0,<2.0.0"
+pydantic = ">=1.9.1,<2.0.0"
```
Comment (arthurio) [inline pyproject.toml:51]: Hmm we do need pydantic `>= 1.10.0` for python 3.11. 🤔 Maybe there is a way to declare that?

**Review by arthurio — COMMENTED** [review]
Looking good, let see what the tests say...

**Review by arthurio — CHANGES_REQUESTED** [review]
Can you tell me why you needed to change the pydantic requirement? Also, it does look like we need to go back to `from typing import Dict, List, ...` to support python 3.8.

**Comment by 10ourto** [comment]
> Can you tell me why you needed to change the pydantic requirement? Also, it does look like we need to go back to `from typing import Dict, List, ...` to support python 3.8.

yep you right we need to fix imports for python 3.8

> pydantic

to be honest I've just downgraded it due to my project requirements, since it has some conflicts

File: pyproject.toml (hunk 1 above)
Comment (arthurio) [reply pyproject.toml:51]: Hmm that's surprising, look at https://github.com/arthurio/fastapi-filter/pull/67

**Comment by 10ourto** [comment]
@arthurio I've updpated PR.
Fixed imports for python 3.8 and also reverted pydantic dependencies back. Feel free to review!

File: README.md [hunk 2]
```diff
@@ -5,10 +5,10 @@
 
 # FastAPI filter
 
-## Compatibilty
+## Compatibility
```
Comment (arthurio) [inline README.md:8]: 👁️

File: docs/index.md [hunk 3]
```diff
@@ -56,7 +56,7 @@ from typing import Optional
 class AddressFilter(Filter):
     street: Optional[str]
     country: Optional[str]
-    city__in: Optional[list[str]]
+    city__in: Optional[List[str]]
```
Comment (arthurio) [inline docs/index.md:59]: I regret my decision when we started supporting 3.9 to update the docs with compatible code. I think that the documentation should show the syntax for the latest supported version. And if I have time to figure out how to show different code examples for different python versions then we can add these. WDYT?

File: examples/fastapi_filter_mongoengine.py [hunk 4]
```diff
@@ -115,15 +115,15 @@ async def on_shutdown() -> None:
...
 
 
-@app.get("/users", response_model=list[UserOut])
+@app.get("/users", response_model=List[UserOut])
 async def get_users(user_filter: UserFilter = FilterDepends(UserFilter)) -> Any:
     query = user_filter.filter(User.objects())
     query = user_filter.sort(query)
     query = query.select_related()
-    return [user.to_mongo() | {"address": user.address.to_mongo()} for user in query]
+    return [{**user.to_mongo(), "address": user.address.to_mongo()} for user in query]
```
Comment (arthurio) [inline examples/fastapi_filter_mongoengine.py:123]: Good catch, I forgot about that nicety... Definitely not missing 3.8 😅

**Review by arthurio — COMMENTED** [review]
Thanks for the updates! Lets see if the tests pass and I'll probably merge/release it this week.

**Comment by 10ourto** [comment]
@arthurio reverted everything related to docs examples back

**Review by arthurio — APPROVED** [review]

**Comment by arthurio** [comment]
Thanks a lot @10ourto !

**Comment by 10ourto** [comment]
@arthurio let me know please when next release is planned?
it could be nice to have new minor release with python 3.8 support

**Pull request merged**